### ISSUE TITLE
Update to v8.0.6 throws error when running functional tests

### ISSUE THREAD

**Issue by brotkrueml** [TYPO3/testing-framework#512]
### What are you trying to achieve?

I updated the package from v8.0.5 to v8.0.6 on a project. I expect that the tests ar running like before.

### What do you get instead?

When running the functional tests on project level, I am getting error for all tests. phpunit v10 does not show any details on that error (displayDetailsOnTestsThatTriggerErrors is set to true).

<del>"Database credentials for tests are neither set through environment variables, and can not be found in an existing LocalConfiguration file"</del> (was another error on my side)

### How to reproduce the issue?

Relevant seems to be running the functional tests on a project level.

1. Update to version 8.0.6
2. Run functional tests

### Additional information you would like to provide?

I checked the commits between v8.0.5 and v8.0.6, the problem occurs with #507.

### Specify some data of the environment

- TYPO3 testing framework version: 8.0.6
- TYPO3 version: 12.4.6
- TYPO3 installation type: Composer
- PHP version: 8.2
- Web server used + version: -
- Operating system: Linux Mint

**Comment by sbuerk** [comment]
I'm curious about that. The ordering should not have an impact on that.
Asking for some more context first, will try to reproduce this later in
some way.

Do you use the environment variables to define the credentials for the tests,
or do you use the fallback to the base installation credentials (settings.php).

Do all extensions have a composer.json (and a ext_emconf.php) ? 

Do all extension have a require to the core (any core sysext) in any way ?

**Comment by brotkrueml** [comment]
The extensions have a composer.json where `typo3/cms-core` and other sysext extensions are required when appropriate.

I also added
```php
   protected array $coreExtensionsToLoad = [
        'adminpanel',
        'backend',
        'belog',
        'beuser',
        'core',
        'dashboard',
        'extbase',
        'extensionmanager',
        'filelist',
        'fluid',
        'fluid_styled_content',
        'form',
        'frontend',
        'info',
        'install',
        'linkvalidator',
        'lowlevel',
        'reactions',
        'recycler',
        'reports',
        'rte_ckeditor',
        'scheduler',
        'seo',
        'setup',
        'tstemplate',
        'viewpage',
    ];
```
but same behaviour.

Sadly, phpunit v10 does not show any error: I assume the error is not happening in the test but during the bootstrap phase of the testing framework.

**Comment by brotkrueml** [comment]
Okay, when using `testdox="false"` the errors are revealed:

---

TYPO3\TestingFramework\Core\Exception: Package "xxx" depends on package "kuria/url" which does not exist.
"kuria/url" is a Composer package configured in the extension's composer.json file.
  
Stacktrace:
```
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/PackageCollection.php:256
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/PackageCollection.php:226
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/PackageCollection.php:117
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/PackageCollection.php:96
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/Testbase.php:620
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:358
/home/chris/Projects/yyy/packages/xxx/Tests/Functional/Zzz/ZzzTest.php:39
```

Line 39 is a call to `parent::setup()`.

Extension XXX has extension AAA loaded via `$testExtensionsToLoad` which requires `kuria/url`via composer.json.

---
  
RuntimeException: The requested database connection named "Default" has not been configured.
  
Stacktrace:
```
/home/chris/Projects/marketplace/vendor/typo3/cms-core/Classes/Database/ConnectionPool.php:142
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/Testbase.php:741
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:274
/home/chris/Projects/yyy/packages/xxx/Tests/Functional/Zzz/ZzzTest.php:39
```

Line 39 is a call to `parent::setup()`.

---

tearDown() check: A dangling error handler has been found. Use restore_error_handler() to unset it.

Stacktrace:
```
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/BaseTestCase.php:57
/home/chris/Projects/marketplace/vendor/typo3/testing-framework/Classes/Core/Functional/FunctionalTestCase.php:413
```

**Comment by sbuerk** [comment]
@brotkrueml Thanks for adding addional information to this. It's not the same, but an issue out of the same area (related) to #510.

**Comment by sbuerk** [comment]
This issue was the same as #510 and has been fixed with #519.
Already released as `8.0.7`.
